content/docs/command-reference/exp/apply.md

```diff
@@ -14,20 +14,17 @@ positional arguments:
 
 ## Description
 
-Restores the results of a given `experiment` (name or hash are accepted, see
-`dvc exp run` for details) into the workspace. This means changing the
-appropriate [metafiles](/doc/user-guide/project-structure),
-<abbr>parameter</abbr> files, <abbr>metrics</abbr>, <abbr>plots</abbr>, and
-corresponding DVC-tracked data.
+Restores a given `experiment` (name or hash are accepted, see `dvc exp run` for
+details) into the workspace. This means changing all files (code, data,
+<abbr>parameters</abbr>, <abbr>metrics</abbr>, etc.) needed to reflect the
+experiment conditions and results in the <abbr>workspace</abbr>.
 
-> This is similar to `dvc checkout`, but for `dvc experiments`.
+This command is typically used after finding a target experiment with
+`dvc exp show` or `dvc exp diff`, and before committing it to Git (making it
+[persistent](/doc/user-guide/experiment-management#persistent-experiments)).
 
-This is typically used after using `dvc exp show` or `dvc exp diff` to find the
-best experiment, and before committing to Git in order to make it
-[persistent](/doc/user-guide/experiment-management#persistent-experiments).
-
-Note that this command will fail if the target `experiment` was not derived from
-the current Git commit.
+Note that this command may fail if the current Git commit (`HEAD`) has changed
+significantly from the time the `experiment` was run.
```

Comment:
It fails outright if your current `HEAD` commit does not match the `HEAD` commit from when you ran the experiment.

It may also fail if there are git conflicts between changes you have in your workspace and changes in the experiment you want to apply. We try to automatically resolve the conflicts by preferring the changes in the applied experiment, but sometimes git may still fail to auto-resolve them.

Reply:
Updated in 8762f974. PTAL

> We try to automatically resolve the conflicts by preferring the changes in the applied experiment

Meaning `git merge --theirs`, right @pmrowla ? I think it's not considered a conflict until after Git can't merge it so I went with the phrase "if they conflict", for now.

> Where  they have moved their HEAD ... and then try to exp apply some experiment

Does/should the CLI print a HINT to try `exp branch` instead?

Reply:
No, it will give an error along the lines of "cannot apply `exp-...` onto commit `<user's current HEAD>`. Experiment is derived from `<original experiment HEAD>`"

Reply:
> exp branch is for a different use case than exp apply

OK @pmrowla. Would you mind checking the bullet list in the latest version to confirm that it covers the use case(s)? Thanks